tests/common/http_mocked_fixtures.py

```diff
@@ -0,0 +1,203 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+HTTP-mocked pytest fixtures for unit tests.
+
+HTTP-mocked pytest fixtures can be used in cases where the zhmcclient mock
+support is not implemented, to mock at the level of the HTTP requests and
+responses instead.
+
+Example use in a pytest test function that invokes a (hypothetical) method
+``Cpc.xyz()``:
+
+.. code-block:: python
+
+    from tests.common.http_mocked_fixtures import http_mocked_session
+    from tests.common.http_mocked_fixtures import http_mocked_cpc_dpm
+
+
+    def test_cpc_xyz(http_mocked_cpc_dpm):
+
+        uri = http_mocked_cpc_dpm.uri + '/operations/xyz'
+
+        rm_adapter = requests_mock.Adapter(case_sensitive=True)
+        with requests_mock.mock(adapter=rm_adapter) as m:
+
+            m.post(uri, status_code=200)
+
+            result = http_mocked_cpc.xyz()
+
+            assert rm_adapter.called
+            request_body = rm_adapter.last_request.json()
+            assert request_body == ...
+            assert result == ...
+"""
+
+import pytest
+import requests_mock
+import zhmcclient
+
+
+@pytest.fixture(
+    scope='module'
+)
+def http_mocked_session(request):  # noqa: F811
+    # pylint: disable=unused-argument
+    """
+    Pytest fixture representing a HTTP-mocked zhmcclient session that is logged
+    on.
+
+    A test function parameter using this fixture resolves to a
+    :class:`~zhmcclient.Session` object that is logged on.
+    """
+
+    session = zhmcclient.Session(
+        host='test-hmc', userid='test-user', password='test-pwd',
+        verify_cert=False)
+
+    with requests_mock.mock() as m:
+        # Because logon is deferred until needed, we perform it
+        # explicitly in order to keep mocking in the actual test simple.
+        m.post('/api/sessions', status_code=200, json={
+            'api-session': 'test-session-id.1',
+            'notification-topic': 'test-obj-topic.1',
+            'job-notification-topic': 'test-job-topic.1',
+            'api-major-version': 4,
+            'api-minor-version': 10,
+            'password-expires': 90,
+        })
+        session.logon()
+
+    yield session
+
+    with requests_mock.mock() as m:
+        m.delete('/api/sessions/this-session', status_code=204)
+        session.logoff()
+
+
+@pytest.fixture(
+    scope='module'
+)
+def http_mocked_cpc_dpm(request, http_mocked_session):  # noqa: F811
+    # pylint: disable=unused-argument,redefined-outer-name
+    """
+    Pytest fixture representing a HTTP-mocked CPC in DPM mode.
+
+    A test function parameter using this fixture resolves to a
+    :class:`~zhmcclient.Cpc` object that is in DPM mode but has only a minimal
+    set of properties in the object (those that are returned with the 'List
+    CPCs' operation).
+    """
+
+    client = zhmcclient.Client(http_mocked_session)
+
+    with requests_mock.mock() as m:
```

Comment:
Would it make sense that the values you are passing here would be arguments (with defaults) to this function?